tools/pull-request-labeler/src/utils/split-message-into-components.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * @license
+ * Copyright 2019 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CommitMessage, CommitType } from '../interfaces/commit-message';
+
+const expression = /(?<type>feat|fix|docs|style|refactor|perf|test|build|chore|barista)(?<component>\(.*?\))?:(?<message>.*)/i;
+/**
+ * Splits the commit message string into its own components. This helps for
+ * further processing of the commit messages.
+ * @param original Original string of the commit message.
+ */
+export function splitStringIntoCommitMessage(original: string): CommitMessage {
+  const matching = original.match(expression);
+
+  if (!matching) {
+    throw new Error(`Message was not parsable: ${original}`);
+  }
+
+  const issueNumber = ((matching.groups && matching.groups.issue) || '').trim();
+  const type = (
+    (matching.groups && matching.groups.type) ||
+    ''
+  ).trim() as CommitType;
+  const componentMatch = (
+    (matching.groups && matching.groups.component) ||
+    ''
+  ).trim();
+  const components = componentMatch
+    .replace('(', '')
+    .replace(')', '')
+    .split(',')
+    .map(component => (component || '').trim())
+    .filter(Boolean);
+
+  const message = ((matching.groups && matching.groups.message) || '').trim();
```

Comment:
Does this message contain the body of the commit as well? @thomaspink mentioned that the breaking change notice should be added to the body from now on, due to the release changelog generator that takes it only from the body

Reply:
This is a good point, i think that it checks for the whole commit message, but let me double check.

Reply:
Examples: 
```
feat(pencil): Add 'graphiteWidth' option
```
```
fix(graphite): stop graphite breaking when width < 0.1

Closes #28
```
```
perf(pencil): Remove graphiteWidth option

BREAKING CHANGE: The graphiteWidth option has been removed. The default graphite width of 10mm is always used for performance reason.
```
```
revert: feat(pencil): Add 'graphiteWidth' option

This reverts commit 667ecc1654a317a13331b17617d973392f415f02.
```

Reply:
@thomaspink I have already extend this so breaking changes are being picked up in the body.  See line 54:56
Can you explain which changes you wanted?